### PR TITLE
Added backend metrics

### PR DESCRIPTION
**Pull request by vrakesh**
*Issue #, if available:*
Added backend metrics, with tests.
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: mms/model_service_worker.py [hunk 1]
```diff
@@ -442,6 +447,23 @@ def run_server(self):
...
+    Parameters
+    ----------
+    metrics: Dictionary
+    A dictionary of all metrics, when key is metric_name
+    value is a metric object
+    """
+    from mms.metric import MetricEncoder
+    cur_time = datetime.datetime.utcnow().strftime('%Y-%m-%d %H:%M:%SZ')
+    metrics['time'] = cur_time
+    print('[METRICS]')
```
Comment (vdantu) [inline mms/model_service_worker.py]: Add a "sys.stdout.flush()" at the end of all prints.

File: mms/metric.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        ----------
+        value : int, float
+           Can be integer or float
+        unit: str
+            unit can be one of ms, percent, count, MB, GB or a generic string
+        metric_method: str
+           useful for defining different operations, optional
+        """
+        self.unit = unit
+        if unit  in list(MetricUnit.units.keys()):
```
Comment (vdantu) [inline mms/metric.py]: How would users be able to use "ms", "s" etc? There seems to be a potential for typo errors from customers. Should we give some sort of enum for the custom class to use?

File: mms/metrics.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    """
+    Class for creating, modifying different metrics. And keep them in a dictionary
+    """
+
+    def __init__(self, req_id_map, model_name):
+        """
+        Initialize metrics map,model name and request map
+        """
+        self.metrics = OrderedDict()
+        self.req_id_map = req_id_map
```
[the remark this answers is not in the document]
Comment (vrakesh) [reply mms/metrics.py]: sure will do

File: mms/metrics.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        self.metrics[model_name] = OrderedDict()
+        if req_id_map != None:
+            for req_id in req_id_map.values():
+                self.metrics[model_name][req_id] = OrderedDict()
+        # When request id is not given it goes to ALL dimension
+        self.metrics[model_name]['ALL'] = OrderedDict()
+        self.metrics[model_name]['ERROR'] = OrderedDict()
```
[the remark this answers is not in the document]
Comment (vrakesh) [reply mms/metrics.py]: Agreed will move error metrics a level up

File: mms/metrics.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+            request id
+        unit: str
+            unit of metric
+        metric_method: str, optional
+            indicates type of metric operation if it is defined
+        """
+        # Create a metric object
+        self.metrics[self.model_name][req_id][name] = Metric(value, unit, metrics_method)
+
+    def _get_req(self, idx):
```
[the remark this answers is not in the document]
Comment (vrakesh) [reply mms/metrics.py]: The default is handled in the level from which this is called, handling it here would be redundant

File: mms/metrics.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+            request id index
+        reverse: boolean, optional
+            by default counters are forward counters, set true to
+            get a decrementing counter
+        """
+        req_id = self._get_req(idx)
+        if name not in self.metrics[self.model_name][req_id]:
+            self._add(name, value, req_id, unit, 'counter')
+            return
+        self.metrics[self.model_name][req_id][name].update(value, reverse)
```
[the remark this answers is not in the document]
Comment (vrakesh) [reply mms/metrics.py]: Yes it is the decrement functionality

**Pull request closed without merging**